packages/casting/src/defaults.js

```diff
@@ -21,7 +21,7 @@ export const DEFAULT_KEEP_POLLING_SECONDS = 5;
  * @param {number} ms
  * @returns {Promise<void>}
  */
-export const delay = ms => new Promise(resolve => setTimeout(resolve, ms));
+export const delay = ms => scheduler.wait(ms);
```

Comment:
This change in particular strikes me as ill-advised; it's non-testing code that could run in a production context.